sematic/db/migrations/20221025201847_add_external_exception_json_field.sql

```diff
@@ -0,0 +1,6 @@
+-- migrate:up
+ALTER TABLE runs ADD COLUMN external_exception_json JSONB;
```

Comment:
Is `external_exception_json` the right name? Is it "external" as in external to the code?
How about `infrastructure_exception_json`?

Reply:
Also, can you remind me the rationale for storing it in a separate field as opposed to re-using `exception_json`?

Reply:
During development I had initially named it "infrastructure", but I reasoned that that name is more restrictive and only applies to cases where we do use underlying infrastructure and it alone failed.

"External" is more generic and potentially covers more sources, such as non-infrastructure network errors while communicating with the external infrastructure. We could (I think we should) rebrand the cancelation signal as an external error, as it does not originate in the code.

We need both of them because we could get both types of failure at the same time, and this is golden information when doing debugging. If we swallow one in favor of the other, we might well in fact be swallowing the root actual cause, and only leaving the other one as a manifestation.